packages/kit/src/runtime/server/cookie.js

```diff
@@ -15,9 +15,35 @@ const cookie_paths = {};
  */
 export function get_cookies(request, url, options) {
 	const header = request.headers.get('cookie') ?? '';
-
 	const initial_cookies = parse(header);
 
+	const normalized_url = normalize_path(
+		// Remove suffix: 'foo/__data.json' would mean the cookie path is '/foo',
+		// whereas a direct hit of /foo would mean the cookie path is '/'
+		has_data_suffix(url.pathname) ? strip_data_suffix(url.pathname) : url.pathname,
+		options.trailing_slash
+	);
+	// Emulate browser-behavior: if the cookie is set at '/foo/bar', its path is '/foo'
+	const default_path = normalized_url.split('/').slice(0, -1).join('/') || '/';
```

Comment:
```suggestion
	const default_path = normalized_url.split('/').slice(0, -1).join('/') ?? '/';
```

Reply:
`anything.join('/')` is a `string`. `'' || '/' === '/'`, but `'' ?? '/' === ''`, which is incorrect